### ISSUE TITLE
Unexpected error message when incorrect subcmd supplied

### ISSUE THREAD

**Issue by steven-cutting** [mahmoud/face#6]
## Issue

When an incorrect subcommand is supplied, the error says that there were `unexpected positional arguments`.

Based on my understanding, positional arguments are not allowed at levels in which there exist further subcommands. So, I believe the error should say something along the lines of `incorrect subcommand provided`.

## Reproduce

```python
from face import Command

cmd = Command(lambda: print("foo"), "foo")
subcmd = Command(lambda: print("bar"), "bar")
subcmd.add(Command(lambda: print("baz"), "baz"))
cmd.add(subcmd)

cmd.run(argv=["foo", "bar", "ba"])  # ba should be baz
```

Results in the following error:

```
error: foo bar: unexpected positional arguments: ['ba']
```

---

If you agree with the above, let me know and I can make it happen.

**Comment by mahmoud** [comment]
Very true! If that level contains any subcommands then yes, absolutely, we should say invalid subcommand. And ideally we'd use [a bit of difflib](https://docs.python.org/2/library/difflib.html#difflib.get_close_matches) to get some suggestions ;)

**Comment by mahmoud** [comment]
Fixed in b736b04d19b0a76a6e0ac598c898f8dbfed60c23!
